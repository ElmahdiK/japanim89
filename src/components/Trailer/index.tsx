'use client';

interface TrailerProps {
  trailerUrl: string;
}

export const defaultYoutubeURL = 'https://www.youtube.com/embed';

export default function Trailer({ trailerUrl }: TrailerProps) {
  return (
    <iframe
      className='h-96 w-full'
      src={`${defaultYoutubeURL}/${trailerUrl}`}
      title='YouTube video player'
      frameBorder='0'
      allow='accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share'
      allowFullScreen
    />
  );
}
