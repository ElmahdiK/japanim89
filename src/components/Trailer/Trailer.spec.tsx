import { render, screen } from '@testing-library/react';
import Trailer, { defaultYoutubeURL } from '.';

describe('test button theme mode', () => {
  it('renders the trailer unchanged', () => {
    const { container } = render(<Trailer trailerUrl='shrek' />);
    expect(container).toMatchSnapshot();
  });

  it('displays the iframe with the right trailer url', () => {
    const trailerEmbed = 'shrek3';
    render(<Trailer trailerUrl={trailerEmbed} />);
    const iframeElement = screen.getByTitle('YouTube video player');
    expect(iframeElement).toBeInTheDocument();
    expect(iframeElement).toHaveAttribute(
      'src',
      `${defaultYoutubeURL}/${trailerEmbed}`
    );
  });
});
