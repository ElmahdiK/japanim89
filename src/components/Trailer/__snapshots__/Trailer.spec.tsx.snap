// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test button theme mode renders the trailer unchanged 1`] = `
<div>
  <iframe
    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
    allowfullscreen=""
    class="h-96 w-full"
    frameborder="0"
    src="https://www.youtube.com/embed/shrek"
    title="YouTube video player"
  />
</div>
`;
